src/mds/CInode.cc

```diff
@@ -5133,6 +5135,151 @@ void CInode::maybe_export_pin(bool update)
   }
 }
 
+class C_Inode_Ephemeral_Distributed_Update : public MDSLogContextBase {
+protected:
+  CInode *in; 
+  MDSRank *get_mds() override {return in->mdcache->mds;}
+  void finish(int r) override {
+    in->is_export_ephemeral_distributed = true;
+  }    
+
+public:
+  C_Inode_Ephemeral_Distributed_Update(CInode *i) : in(i) {}
+};
+
+class C_Inode_Ephemeral_Random_Update : public MDSLogContextBase {
+protected:
+  CInode *in;
+  MDSRank *get_mds() override {return in->mdcache->mds;}
+  void finish(int r) override {
+    in->is_export_ephemeral_random = true;
+  }
+
+public:
+  C_Inode_Ephemeral_Random_Update(CInode *i) : in(i) {}
+};
+
+void CInode::maybe_export_ephemeral_pin(bool update)
+{
+  //If both the configs aren't set then return
+  if (!g_conf().get_val<bool>("mds_export_ephemeral_random")
+      && !g_conf().get_val<bool>("mds_export_ephemeral_distributed"))
```

Comment:
These configs need cached by in `MDCache`. Fetching these are relatively expensive. See `handle_conf_change`.